### ISSUE TITLE
Scroll View height being calculated incorrectly when keyboard active

### ISSUE THREAD

**Issue by jstheoriginal** [zendesk/zendesk_sdk_chat_ios#89]
When the keyboard is active, there is a gap equal to the height of the tab bar between the scrollview and the keyboard. Our custom tab bar is 54 pts high and I measured the pixel height of the gap in photoshop and it is also 54 pts.

Any chance this can be changed to calculate correctly?

![simulator-screen-shot-may-26 -2016 -10 21 01-am](https://cloud.githubusercontent.com/assets/5117473/15580172/40c81b76-232d-11e6-9f68-a29ba0d8e57d.png)

**Comment by jstheoriginal** [comment]
I bet this is also causing the jumping in this issue by misreading the view's height when it redraws for the second line. https://github.com/zendesk/zendesk_sdk_chat_ios/issues/87

**Comment by jstheoriginal** [comment]
This is also happening in the actual chat screen as well...this will need to be fixed before this goes live in our app as it looks quite bad.

![simulator screen shot may 26 2016 10 39 11 am](https://cloud.githubusercontent.com/assets/5117473/15580615/f591c06a-232e-11e6-8e06-de5359de7d8c.png)

**Comment by oarrabi** [comment]
Hey @jstheoriginal,

It seems the issue is not visible with normal tab bars. 

Can you please provide more info on how you are creating your tab bar? 
Does this issue happen on all iOS versions?
Simulator and Device?
and how you are starting the chat?

It would be great if you could provide a sample project to reproduce it :)

Thanks for the info.

**Comment by jstheoriginal** [comment]
@oarrabi Hey Omar.  Here's some more info. I'll work on providing a sample project if what I said below doesn't make it obvious what the issue is:

> Can you please provide more info on how you are creating your tab bar?

Nothing too crazy here. On our root view controller, we are creating the tab bar manually in code with a custom UIView container to hold the custom tab bar/tabs. Each tab contains a UINavigationController with its custom UIViewController as its root. The tab bar is opaque while the navigation bar is transparent.

> Does this issue happen on all iOS versions?

Yes...tested on 9.3 and 8.1.

> Simulator and Device?

Yes both.

> and how you are starting the chat?

The chat is being started like this from within the tab's UIViewController:

```swift
    @IBAction private func contactSupportButtonTapped() {
        guard let navController = navigationController else {
            fatalError("HelpViewController is not inside a navigation controller")
        }
        if let customer = customer {
            ZDCChat.updateVisitor { visitor in
                visitor.phone = customer.phoneNumber
                visitor.name = customer.name
                visitor.email = customer.email
            }
            ZDCChat.startChatIn(navController, withConfig: nil)
        }
    }
```

Thanks! Let me know if you still need a sample project.

**Comment by oarrabi** [comment]
Hello @jsthteoriginal,

As for the reply in #93, thanks so much for providing a demo, this issue and #93 are fixed and will be published in the next release.

Thanks for your help :)

**Comment by ndobri** [comment]
@jstheoriginal this was fixed in [Chat SDK 1.3](https://github.com/zendesk/zendesk_sdk_chat_ios/releases/tag/1.3.0.1). Closing this for now, but if the issue persist please reopen this thread and let us know.
